### PR TITLE
Fill the section "Maven Level 2" 

### PR DESCRIPTION
**Pull request by dduportal**
Signed-off-by: dduportal <1522731+dduportal@users.noreply.github.com>

### CODE REVIEW

File: content/chapitres/maven-advanced.adoc [hunk 1]
```diff
@@ -1,21 +1,277 @@
...
+<build>
+  <name>${ensg.student.name}</name>
+</build>
+----
+
+== Exercice : Définir la plateforme d'exécution
+
+*But* : la commande `mvn compile` doit fonctionner sans erreur, et produire un fichier `Hello.class` dans `./target/**`
+
+. Modifiez le fichier `pom.xml` pour ajouter un bloc `<properties>` et définissez la valeur de la directive `project.build.sourceEncoding` à `UTF-8` (résolution du warning).
```
Comment (jlevesy) [inline content/chapitres/maven-advanced.adoc]:
```suggestion
. Modifiez le fichier `pom.xml` pour ajouter un bloc `<properties>` et définissez la valeur de la propriété `project.build.sourceEncoding` à `UTF-8` (résolution du warning).
```

wip wip wip !

File: content/chapitres/maven-advanced.adoc [hunk 2]
```diff
@@ -1,21 +1,277 @@
...
+# "-cp" == "classpath" (Chemin vers les classes Java "compilées")
+java -cp ./target/classes/ Hello
+# Argument "Hello" == classe qui contient la méthode statique "main"
+----
+
+== Maven : Dépôts d'Artefacts
+
+Maven récupère les dépendances (et plugins) dans des dépôts d'artefacts
+
+(🇬🇧 Artefacts Repositories) qui sont de 3 types :
```
Comment (jlevesy) [inline content/chapitres/maven-advanced.adoc]:
```suggestion
(🇬🇧 Artifacts Repositories) qui sont de 3 types :
```

wip wip wip `Artifacts` wip wip wip wip

File: content/chapitres/maven-advanced.adoc [hunk 3]
```diff
@@ -1,21 +1,277 @@
...
+
+* *Central* : un dépôt géré par la communauté - https://repo.maven.apache.org[]
+* *Remote* : un dépôt de votre organsation, similaires à un remote GitHub, hébergé par vos soins
```
Comment (jlevesy) [inline content/chapitres/maven-advanced.adoc]:
```suggestion
* *Remote* : un dépôt de votre organisation, similaires à un remote GitHub, hébergé par vos soins
```

wip wip wip

File: content/chapitres/maven-advanced.adoc [hunk 4]
```diff
@@ -1,21 +1,277 @@
...
+  </dependency>
+</dependencies>
+----
+
+== Exercice avec les dépendances Spring
+
+=> C'est à vous. Ajoutez le bloc précédent dans votre `pom.xml`
+
+* Exécutez la commande `mvn clean compile`
+* Explorer le contenu du dossier `$HOME/.m2` (écriture équivalente à `~/.m2`)
```
Comment (jlevesy) [inline content/chapitres/maven-advanced.adoc]:
```suggestion
* Explorez le contenu du dossier `$HOME/.m2` (écriture équivalente à `~/.m2`)
```

wip wip wip wip 'z' 'r'

File: content/chapitres/maven-advanced.adoc [hunk 5]
```diff
@@ -1,21 +1,277 @@
...
+ls -l ~/.m2/repository/org/apache/maven/plugins/
+
+* Supprimez le dossier `~/.m2/` et relancez la commande `mvn clean compile`
+
+== Solution avec les dépendances Spring
+
+* Le dépôt local `.m2` :
+** Contient tout : dépendances et artefacts
+** Respecte la structure des groupid, artifactId et version
+** Agit comme un "cache" local
```
Comment (jlevesy) [inline content/chapitres/maven-advanced.adoc]: wip wip wip "mvn install" wip wip wip wip

**Review by jlevesy — COMMENTED** [review]
:eyes: 

wip wip wip wip 🎉 

wip wip wip wip 🤔  ? wip wip wip wip  🚗 ! 

wip wip wip wip 

wip wip !

File: content/chapitres/maven-advanced.adoc [hunk 6]
```diff
@@ -1,23 +1,357 @@
...
+[source,xml]
+----
+<groupId>com.mycompany.app</groupId>
+<artifactId>my-app</artifactId>
+<version>1.0-SNAPSHOT</version>
+----
+
+== Exercice : Maven From Scratch
+
+=> C'est à vous dans link:https://gitpod.io#https://github.com/cicd-lectures/demoapp[l'environnement GitPod,window="_blank"]
```
Comment (jlevesy) [inline content/chapitres/maven-advanced.adoc]: Shouldn't we point them to their current workspace instead ?

File: content/chapitres/maven-advanced.adoc [hunk 7]
```diff
@@ -1,23 +1,357 @@
...
+
+* Créez un projet vide pour Maven :
+[source,bash]
+mkdir -p /workspace/mvn-level2 && cd /workspace/mvn-level2
+
```
Comment (jlevesy) [inline content/chapitres/maven-advanced.adoc]:
```suggestion
mkdir -p /workspace/mvn-level2 && cd /workspace/mvn-level2 && mkdir -p src/main/java

```

File: content/chapitres/maven-advanced.adoc [hunk 8]
```diff
@@ -1,23 +1,357 @@
...
+== Dépendances Maven
+
+Pour spécifier les dépendances :
+
+* Il faut utiliser la balise `<dependencies>`,
+* ... qui est une collection de dépendances (balise `<dependency>` - quelle surprise !),
+* .. chaque dépendance étant défini par un trio `<groupId>`, `<artifactId>` et `<version>` (que de surprises...)
+
+Pour les plugins c'est la même idée (`<plugin>` -> `<plugin>` -> `<groupId>`, `<artifactId>`, `<version>`)
```
Comment (jlevesy) [inline content/chapitres/maven-advanced.adoc]:
```suggestion
Pour les plugins c'est la même idée (`<plugins>` -> `<plugin>` -> `<groupId>`, `<artifactId>`, `<version>`)
```

File: content/chapitres/maven-advanced.adoc [hunk 9]
```diff
@@ -1,23 +1,357 @@
...
+* *But*: Produire un artefact JAR dont le nom est constant
+
+* Toujours dans link:https://gitpod.io#https://github.com/cicd-lectures/demoapp[l'environnement GitPod,window="_blank"],
+exécutez la commande `mvn package`
+
+* Quel est le nom de l'artefact généré ? Est-il constant ?
+** (SPOILER: 🙅🏽‍♀️)
+
+* En utilisant la documentation de référence link:https://maven.apache.org/pom.html#the-basebuild-element-set[],
+adaptez votre `pom.xml` afin que le fichier généré se nomme *toujours$  `hello.jar`.
```
Comment (jlevesy) [inline content/chapitres/maven-advanced.adoc]:
```suggestion
adaptez votre `pom.xml` afin que le fichier généré se nomme *toujours*  `hello.jar`.
```

File: content/chapitres/maven-advanced.adoc [hunk 10]
```diff
@@ -1,23 +1,357 @@
...
+      </plugin>
+  </plugins>
+</build>
+----
+
+== Exercice : Couverture de code avec Jacoco
+
+* *But* : Générer un/des rapport(s) de couverture de code
+
+* Reprenez la branche `full-path` de `demoapp`, qui contient des tests unitaires et d'intégration
```
Comment (jlevesy) [inline content/chapitres/maven-advanced.adoc]:
```suggestion
* Reprenez la branche `full-maven` de `demoapp`, qui contient des tests unitaires et d'intégration
```

**Review by jlevesy — COMMENTED** [review]
👀

File: content/chapitres/maven-advanced.adoc (hunk 7 above)
Comment (dduportal) [reply content/chapitres/maven-advanced.adoc]:
```suggestion
mkdir -p /workspace/mvn-level2/src/main/java && cd /workspace/mvn-level2

```

**Review by jlevesy — APPROVED** [review]

**Pull request merged**